File: packages/dotnet/Ygo74.AgentRuntime/Domains/Auth/ApiKeyAuthenticator.cs
namespace Ygo74.AgentRuntime.Domains.Auth;

public static class ApiKeyAuthenticator
{
    public static IDictionary<string, string> Authenticate(string apiKey, Func<string, IDictionary<string, string>?> resolver)
    {
        var user = resolver(apiKey);
        if (user is null)
        {
            throw new ArgumentException("invalid api key", nameof(apiKey));
        }

        user["authType"] = "api_key";
        return user;
    }
}

File: packages/java/ygo74-agent-runtime/src/main/java/com/ygo74/agentruntime/domains/auth/ApiKeyAuthenticator.java
package com.ygo74.agentruntime.domains.auth;

import java.util.Map;
import java.util.function.Function;

public final class ApiKeyAuthenticator {
    private ApiKeyAuthenticator() {}

    public static Map<String, String> authenticate(String apiKey, Function<String, Map<String, String>> resolver) {
        var user = resolver.apply(apiKey);
        if (user == null) {
            throw new IllegalArgumentException("invalid api key");
        }
        user.put("authType", "api_key");
        return user;
    }
}
